src/Microsoft.Diagnostics.Runtime/src/Builders/RuntimeBuilder.cs

```diff
@@ -687,7 +687,20 @@ private void FlushDac()
             }
         }
 
-        ulong IRuntimeHelpers.GetMethodDesc(ulong ip) => _sos.GetMethodDescPtrFromIP(ip);
+        ulong IRuntimeHelpers.GetMethodDesc(ulong ip)
+        {
+            ulong md = _sos.GetMethodDescPtrFromIP(ip);
+            if (md == 0)
+            {
+                if (!_sos.GetCodeHeaderData(ip, out CodeHeaderData codeHeaderData))
+                    return 0;
+
+                if ((md = codeHeaderData.MethodDesc) == 0)
+                    return 0;
```

Comment:
```suggestion
                md = codeHeaderData.MethodDesc;
```

Reply:
The current implementation is more explicit in term of error handling (0 is supposed to be an error)